src/components/ScriptExplorer/IndirectSignatureImporter.jsx

```diff
@@ -0,0 +1,256 @@
+        />
+      );
+    }
+    return (
+      <Box mt={2}>
+        <Box mt={2}>
+          {this.renderDeviceConfirmInfo()}
+          <FormGroup>
+            <Signer
+              setError={this.setError}
+              hasError={this.hasBIP32PathError()}
+              onReceive={this.onReceive}
+              onReceivePSBT={this.onReceivePSBT}
+              interaction={this.interaction()}
+              setActive={this.setActive}
+              disableChangeMethod={disableChangeMethod}
+              extendedPublicKeyImporter={extendedPublicKeyImporter}
+            />
+
+            <FormHelperText error>{signatureError}</FormHelperText>
+
+            <InteractionMessages
+              messages={interaction.messagesFor({ state: status })}
+            />
+          </FormGroup>
+        </Box>
+      </Box>
+    );
+  };
+
+  setActive = () => {
+    this.setState({ status: ACTIVE });
+  };
+
+  onReceive = (signature) => {
+    const { validateAndSetSignature, enableChangeMethod } = this.props;
+    this.setState({ signatureError: "" });
+    if (enableChangeMethod) {
+      enableChangeMethod();
+    }
+    validateAndSetSignature(signature, (signatureError) => {
+      this.setState({ signatureError });
+    });
+  };
+
+  onReceivePSBT = (data) => {
+    const { validateAndSetSignature } = this.props;
+    try {
+      // signatureData is one or more sets of signatures that are keyed
+      // based on which pubkey the signatures are signing.
+      const signatureData = this.interaction().parse(data);
+      const signatureSetsKeys = Object.keys(signatureData);
+      const signatures = [];
+
+      // We have a slight issue for the n-ly signed PSBT case
+      // because there is no order to the pubkey: [signature(s)] mapping
+      // returned from `unchained-bitcoin`. it's ok, we have valid signatures, etc.
+      // but truncating information can cause problems down the line in keeping
+      // the validation straightforward.
+
+      // e.g. have a doubly-signed psbt with 3 inputs on the same 2-of-3 multisig address,
+      // so 6 signatures total are returned in signatureData back for 2 pubkeys (3 signatures from each pubkey).
+      // here adding a pubkey set matches that signature array.
+      // [siga1, siga2, siga3, sigb1, sigb2, sigb3]
+      // In the case of multiple 2-of-3 multisig addresses, the pubkeys can get jumbled and there's no longer
+      // a clear break between signature sets based purely on pubkey. need xfp information as well.
+
+      // For now, shove all of the signatures into the same array and return that.
+      signatureSetsKeys.forEach((pubkey) => {
+        signatures.push(...signatureData[pubkey]);
+      });
+
+      this.setState({ signatureError: "" });
+      validateAndSetSignature(signatures, (signatureError) => {
+        this.setState({ signatureError });
+      });
+    } catch (e) {
+      e.errorType = "Coldcard Signing Error";
```

Comment:
I like this idea of an errorType. Brings to mind that it would be good to eventually have some sort of Caravan specific error classes. Obviously something for the future though.